source/extensions/filters/http/jwt_authn/jwt_cache.cc

```diff
@@ -0,0 +1,71 @@
+#include "extensions/filters/http/jwt_authn/jwt_cache.h"
+
+#include "common/common/assert.h"
+
+namespace Envoy {
+namespace Extensions {
+namespace HttpFilters {
+namespace JwtAuthn {
+namespace {
+
+// The default number of entries in JWT cache is 100.
+constexpr int kJwtCacheDefaultSize = 100;
+
+class JwtCacheImpl : public JwtCache {
+public:
+  JwtCacheImpl(bool enable_cache, int cache_size) {
+    if (enable_cache) {
+      // if cache_size is 0, it is not specified in the config, use default
+      if (cache_size == 0) {
+        cache_size = kJwtCacheDefaultSize;
+      }
+      jwt_cache_ =
+          std::make_unique<SimpleLRUCache<std::string, ::google::jwt_verify::Jwt>>(cache_size);
+    }
+  }
+
+  ~JwtCacheImpl() {
+    if (jwt_cache_) {
+      jwt_cache_->clear();
+    }
+  }
+
+  ::google::jwt_verify::Jwt* lookup(const std::string& token) override {
+    if (!jwt_cache_) {
+      return nullptr;
+    }
+    ::google::jwt_verify::Jwt* found_jwt{};
+    SimpleLRUCache<std::string, ::google::jwt_verify::Jwt>::ScopedLookup lookup(jwt_cache_.get(),
+                                                                                token);
+    if (lookup.found()) {
+      found_jwt = lookup.value();
+      ASSERT(found_jwt != nullptr);
+      if (found_jwt->verifyTimeConstraint(absl::ToUnixSeconds(absl::Now())) ==
```

Comment:
use timeSource to get "now",  so it will be easier to write  unit-test.   
Just like line 195 in auehenticator.cc.  this is just for cache hit.  not need to handle  clock_skew

Reply:
Done.